### PR TITLE
Feature/settings Dev setting screen UI

### PR DESCRIPTION
**Pull request by TriDFDev**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,6 +20,8 @@
     "lottie-react-native": "^5.1.5",
     "react": "18.2.0",
     "react-native": "0.71.7",
+    "react-native-circular-progress": "^1.3.8",
+    "react-native-circular-progress-indicator": "^4.4.2",
```
Comment (vinguyen2307) [inline package.json:24]: Don't use too many libraries, You need write your own components

File: src/components/Global/AnimatedChord.tsx [hunk 2]
```diff
@@ -7,23 +7,38 @@ import {
...
   width: number;
   height: number;
   style?: any;
   index?: any;
+  nameChord: string;
 }
-const Chord: React.FC<ChordProps> = props => {
+const ChordAnimation: React.FC<ChordProps> = props => {
   const [chordAnim] = useState(new Animated.Value(0));
   useEffect(() => {
```
Comment (vinguyen2307) [inline src/components/Global/AnimatedChord.tsx:25]: Move useEffect to line 52

File: src/components/Global/ChordChart.tsx [hunk 3]
```diff
@@ -13,7 +13,7 @@ interface Props {
   width?: number
   chord?: Array<string>
   showTuning?: boolean
-  tuning?: Array<string>
+  tuning?: Array<string>| any
```
Comment (vinguyen2307) [inline src/components/Global/ChordChart.tsx:16]: Not use any

File: src/components/Global/ChordChart.tsx [hunk 4]
```diff
@@ -234,7 +234,7 @@ const ChordChart: FunctionComponent<Props> = (props) => {
 
         {// Draw tuning
           showTuning && tuning.length == numStrings &&
-          tuning.map((t, i) => {
+          tuning.map((t: any, i: any) => {
```
Comment (vinguyen2307) [inline src/components/Global/ChordChart.tsx:237]: Not use type any

File: src/screens/ChordGame.tsx [hunk 5]
```diff
@@ -13,22 +13,38 @@ const ChordGame = () => {
...
-    chordList: [{name: 'F'}, {name: 'Fm'}, {name: 'A'}, {name: 'Amaj7'}],
-    scrore: 0.3,
-    title: "4 Base chords"
-  }, {
-    status: false,
-    chordList: [{name: 'C'}, {name: 'Am'}, {name: 'Em'}, {name: 'G'}],
-    scrore: 0,
-    title: "4 Base chords"
-  }];
+  const Data = [
```
Comment (vinguyen2307) [inline src/screens/ChordGame.tsx:16]: Move Data to file constant

File: src/screens/GameDetail.tsx [hunk 6]
```diff
@@ -1,24 +1,28 @@
...
-import {useNavigation} from '@react-navigation/native';
+import {useNavigation, useRoute} from '@react-navigation/native';
 import {RootStackScreenProps} from '../types/navigation/types';
 import {DARKCOLORS} from '../constants/colors';
 import {FONTFAMILY} from '../constants/fonts';
 import {FONTSIZE} from '../constants/sizes';
 import {GENERALSTLE} from '../styles/generalStyle';
-import Chord from '../components/Global/AnimatedChord';
+import ChordAnimation from '../components/Global/AnimatedChord';
 import SelectChord from './SelectChord';
```
Comment (vinguyen2307) [inline src/screens/GameDetail.tsx:11]: Update style import

Import Lib

Import Component

Import constant/styles/ultils

File: src/screens/Setting.tsx [hunk 7]
```diff
@@ -1,15 +1,124 @@
...
-}
+  );
+};
 
-export default Setting
+export default Setting;
 
-const styles = StyleSheet.create({})
\ No newline at end of file
+const styles = StyleSheet.create({
```
Comment (vinguyen2307) [inline src/screens/Setting.tsx:49]: You need create file ***.styles.ts and move styles to ***.styles.ts

File: src/utils/shuffleArray.ts [hunk 8]
```diff
@@ -0,0 +1,8 @@
+export const shuffleArray = (array: string[]) => {
+  const newArray = [...array];
```
Comment (vinguyen2307) [inline src/utils/shuffleArray.ts:2]: Can you explain to me why you code this?

Comment (TriDFDev) [reply src/utils/shuffleArray.ts:2]: I want the answers to change positions, to avoid the player answering by memorizing the position of the answers.

**Pull request merged**